gdsc-web-backend/Controllers/ContactController.cs

```diff
@@ -6,11 +9,30 @@ namespace gdsc_web_backend.Controllers
     [Route("api/[controller]")]
     public class ContactController : ControllerBase
     {
+        private readonly List<ContactModel> _mockContact = new();
+
+        
         [HttpPost]
-        public ActionResult Post()
+        [ProducesResponseType(typeof(ErrorViewModel), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ContactModel), StatusCodes.Status201Created)]
+        public ActionResult<ContactModel> Post([FromBody] ContactModel entity)
         {
-            return StatusCode(201);
+            if (entity is null)
+            {
+                return BadRequest(new ErrorViewModel {Message = "Request has no body"});
+            }
+
+            //create a variable where we return the value of the find function applied on the _mockContact
+            var doesExists = _mockContact.Find(p => p.Id == entity.Id);
+            if (doesExists != null)
+            {
+                return BadRequest(new ErrorViewModel {Message = $"{entity} already exists"});
+            }
+
+            _mockContact.Add(entity);
+
+            entity = _mockContact.Find(e => e == entity);
+            return Created("api/Contact/" + entity!.Id, entity);
```

Comment:
Here we should return only Ok(entity), since contact has not Get method to use for returned url